src/switch_ivr_originate.c

```diff
@@ -1931,6 +1933,43 @@ static void *SWITCH_THREAD_FUNC early_thread_run(switch_thread_t *thread, void *
 				}
 
 				if (!state->ringback->asis) {
+					switch_codec_implementation_t peer_read_impl = { 0 }, write_impl = { 0 };
+
+					if (switch_core_session_get_read_impl(originate_status[array_pos].peer_session, &peer_read_impl) == SWITCH_STATUS_SUCCESS 
+						&& switch_core_session_get_write_impl(state->oglobals->session, &write_impl) == SWITCH_STATUS_SUCCESS) {
+
+						// sync sample rate mismatch for bridge_early_media
+						if ((peer_read_impl.actual_samples_per_second != write_impl.actual_samples_per_second && peer_read_impl.actual_samples_per_second != state->write_frame->codec->implementation->actual_samples_per_second)
```

Comment:
On the first run of `early_thread_run`, what happened when `peer_read_impl` and `write_impl` matches? I think we won't create the `write_codec`

Reply:
Yes. We won't need to recreate the `write_codec` as long as `peer_read_impl` and `write_impl` matches.

`wrtie_codec` is first created in `setup_ringback` function. It'll only be recreated if the `peer_read_impl` changes on the other leg for this condition

Reply:
I just read the code, it was the read_codec that is created initially by this function and you now added/created the write codec.